src/Npgsql/Replication/ReplicationConnection.cs

```diff
@@ -444,6 +444,69 @@ async Task<TimelineHistoryFile> TimelineHistoryInternal(uint tli, CancellationTo
         }
     }
 
+    internal async Task<ReplicationSlotInfo> ReadReplicationSlotInternal(string slotName, CancellationToken cancellationToken = default)
+    {
+        CheckDisposed();
+
+        using var _ = Connector.StartUserAction(cancellationToken, attemptPgCancellation: _pgCancellationSupported);
+
+        var command = "READ_REPLICATION_SLOT " + slotName;
+        Log.Debug($"Executing replication command:{Environment.NewLine}\t{command}", Connector.Id);
+        await Connector.WriteQuery(command, true, cancellationToken);
```

Comment:
Could this method simply call ReadSingleRow? It seems like it could reduce a lot of code here (it would box since it returns `object[]`, but that doesn't seem like a concern here).

Reply:
Yeah, that's an oversight. I already forgot that we have `ReadSingleRow()`.
Thanks for reminding me.

Reply:
Then again `ReadSingleRow()` doesn't work in this case for the same reason it doesn't work for `CreateReplicationSlot()`.
The `restart_lsn` column gets returned as text and `ReadSingleRow()` doesn't know that in this special case it should parse it into a `NpgsqlLogSequenceNumber`.

We could certainly teach it to do some magic based on the field name or the like but that seems like an ugly footgun to me.

Reply:
@Brar I was thinking that ReadReplicationSlotInternal would simply use ReadSingleRow to get the object array, and then ReadReplicationSlotInternal would parse the string into an NpgsqlLogSequenceNumber (as it already does right now). So ReadSingleRow would only be used for the low-level reading stuff - any conversion/parsing can happen later in the method calling it.

Reply:
> I was thinking that ReadReplicationSlotInternal would simply use ReadSingleRow to get the object array, and then ReadReplicationSlotInternal would parse the string into an NpgsqlLogSequenceNumber

:man_facepalming: yes, that would obviously work (also for `CreateReplicationSlot()`).
Sometimes I really don't understand how such obvious things don't come to my mind.